### ISSUE TITLE
[BUG] Performance Issues with SVG Map on Mobile Devices

### ISSUE THREAD

**Issue by ajaynegi45** [ajaynegi45/Uttarakhand-Culture-NewUI#40]
### Is this a new issue?

- [X] I have searched "open" and "closed" issues, and this is not a duplicate.

### Bug Description

The SVG map on the [website](https://uttarakhand-culture.vercel.app/map) is laggy during zoom in/out on older mobile devices. Additionally, on smaller screens (e.g., mobile phones or tablets), the text inside the SVG overlaps, making it difficult to read.

**Requirements**:

- **Zoom Performance**: Improve the zoom in/out speed, especially on older devices, to make it as fast as possible.
- **Text Overlap**: Fix the text overlap issue on small screens.
- **General Improvements**: Any extra enhancements to improve the map are welcome, but do not use any external libraries or frameworks.

Before starting the implementation, please discuss your plan and approach.

### Steps to Reproduce

- Go to https://uttarakhand-culture.vercel.app/map

### Screenshots

# Text Overlap
![Screenshot_2024-10-01-23-10-03-571-edit_com android chrome](https://github.com/user-attachments/assets/b3a808f8-f178-4f50-ae0f-c9aeed9ed065)

**Comment by shashmitha46** [comment]
can you assign me this?

**Comment by ajaynegi45** [comment]
> shashmitha46

yes, you can work on it.

**Comment by ajaynegi45** [comment]
> /assign

Hi @nilesh-fatfatwale,

Thank you for expressing your interest in working on this issue. I'm delighted to inform you that I have assigned this issue to you. Your willingness to contribute to our project is much appreciated.

Before making any contributions, please read the Readme.md file carefully to better understand the project's goals and purpose. This will give you clarity on our mission. Also, make sure to review the Contributing.md file, which contains important guidelines to make the contribution process smoother, especially for newcomers.

Feel free to start working, and if you have any questions or need assistance during the process, please don't hesitate to reach out.
